src/sentry/static/sentry/app/components/asyncComponent.jsx

```diff
@@ -76,16 +76,21 @@ class AsyncComponent extends React.Component {
 
     endpoints.forEach(([stateKey, endpoint, params, options]) => {
       options = options || {};
+      let locationQuery = (this.props.location && this.props.location.query) || {};
+      let paramsQuery = (params && params.query) || {};
+      let query = options.paginate && {...locationQuery, ...paramsQuery};
```

Comment:
could you add a comment to show what keys are flowing through here from the location/params and ending up in the query?